lexer/lexer.cc

```diff
@@ -210,18 +210,131 @@ Token Lexer::parseStringLiteral() {
   // TODO handle raw string/heredoc
   const char *firstPos = mLastPosition;
   const char *lastPos = mLastPosition;
+
   while (mLast != '"') {
-    if (mLast == 0)
+    if (mLast == 0) {
       throw std::runtime_error(fmt::format(
-          "{}: Lexer error: unerminated string literal", mLocation));
+          "{}: Lexer error: unterminated string literal", mLocation));
+    }
+
+    if (mLast == '\\') {
+      // Got escaped character. Handle this string literal separately.
+      return parseStringLiteralWithEscapedCharacters(firstPos);
+    }
+
     readNextChar();
+
     lastPos = mLastPosition;
   }
+
   readNextChar(); // skip past the closing quote
   std::string_view content(firstPos, lastPos - firstPos);
   return Token{TokenType::LitString, mTokenBeginLocation, content};
 }
 
+Token Lexer::parseStringLiteralWithEscapedCharacters(const char *firstPos) {
+
+  std::string result(firstPos, mLastPosition);
```

Comment:
I'd prefer this to be created already in the Zone, w/o heap allocation.
You can use the length of string with escapes as an upper-bound for allocation.

Reply:
Going forward, we'd need a StringBuilder and a Twine class that works and lives in the Zone

Reply:
Good idea about using the Zone, not `std::string`. I can work on `StringBuilder`. By `Twine` you mean something like `llvm::Twine`?

Reply:
Yes. I'm thinking now haw this could be done. It would be useful for qualified names, like `module::type` or `class::method`